mobile/src/reducers/people/people.reducer.ts

```diff
@@ -0,0 +1,162 @@
+
+export interface PeopleState {
+    headDepartment: Department;
+    departments: Department[];
+    departmentsHeads: Employee[];
+    employees: Employee[];
+    departmentsHeadsIds: string[];
+    departmentsBranch: DepartmentsTreeNode[];
+    departmentsTree: DepartmentsTree;
+}
+
+const initState: PeopleState = {
+    headDepartment: null,
+    departments: null,
+    departmentsHeads: null,
+    employees: null,
+    departmentsHeadsIds: null,
+    departmentsBranch: null,
+    departmentsTree: null
+};
+
+function departmentsTreeFor(departments: Department[], employees: Employee[]) {
+    const topLevelDepartment: Department = departments.filter((department) => department.isHeadDepartment === true)[0];
+    const topLevelEmployee: Employee = employees.filter((employee) => employee.employeeId === topLevelDepartment.chiefId)[0];
+    const departmentsTree: DepartmentsTree = {
+        root: {
+            departmentId: topLevelDepartment.departmentId,
+            departmentAbbreviation: topLevelDepartment.abbreviation,
+            departmentChiefId: topLevelDepartment.chiefId,
+            head: topLevelEmployee,
+            parent: null,
+            children: childrenNodes(topLevelDepartment, departments, employees),
+            subordinates: employees.filter((employee) => employee.departmentId === topLevelDepartment.departmentId)
+        }
+    };
+
+    return departmentsTree;
+}
+
+function onlyUnique(value: string, index: number, self: string[]) { 
+    return self.indexOf(value) === index;
+}
+
+function childrenNodes(headDeaprtment: Department, departments: Department[], employees: Employee[]) {
+    var nodes: DepartmentsTreeNode[] = [];
+    const sublings = departments.filter((subDepartment) => subDepartment.parentDepartmentId === headDeaprtment.departmentId);
+    sublings.forEach(department => {
+        const employee: Employee = employees.filter((emp) => emp.employeeId === department.chiefId)[0];
+        const subSublings = departments.filter((subDepartment) => subDepartment.parentDepartmentId === department.departmentId);
+        nodes.push({
+            departmentId: department.departmentId,
+            departmentAbbreviation: department.abbreviation,
+            departmentChiefId: department.chiefId,
+            head: employee,
+            parent: headDeaprtment,
+            children: subSublings.length > 0 ? childrenNodes(department, departments, employees) : null,
+            subordinates: employees.filter((emp) => emp.departmentId === department.departmentId)
+        });
+    });
+
+    return nodes;
+}
+
+export const peopleReducer: Reducer<PeopleState> = (state = initState, action: PeopleActions | LoadEmployeeFinished | LoadDepartmentsFinished) => {
+    switch (action.type) {
+        case 'LOAD_EMPLOYEE_FINISHED':
+            var heads = state.departmentsHeads;
+            if (heads === null) {
+                heads = [];
```

Comment:
Why don't you assign empty array in the init state?

Reply:
👍 addressed